pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -203,6 +203,34 @@ func (builder *ListenerBuilder) buildVirtualInboundListener(env *model.Environme
 	return builder
 }
 
+func (builder *ListenerBuilder) patchListeners(push *model.PushContext) {
+	if builder.node.Type == model.Router {
+		envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_GATEWAY, builder.node, push, builder.gatewayListeners, false)
+		return
+	}
+
+	patchOneListener := func(listener *xdsapi.Listener) *xdsapi.Listener {
+		if listener == nil {
+			return nil
+		}
+		tempArray := []*xdsapi.Listener{listener}
+		tempArray = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_OUTBOUND, builder.node, push, tempArray, true)
+		// temp array will either be empty [if virtual listener was removed] or will have a modified listener
+		if len(tempArray) == 0 {
+			return nil
+		}
+		return tempArray[0]
+	}
+	builder.virtualListener = patchOneListener(builder.virtualListener)
+	builder.virtualInboundListener = patchOneListener(builder.virtualInboundListener)
+	builder.managementListeners = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_INBOUND, builder.node,
+		push, builder.managementListeners, true)
+	builder.inboundListeners = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_INBOUND, builder.node,
+		push, builder.inboundListeners, false)
+	builder.outboundListeners = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_INBOUND, builder.node,
```

Comment:
ditto
outbound listeners are applying EnvoyFilter_SIDECAR_INBOUND patches?

Reply:
Oh! good catch.. that is a typo.

Reply:
Let me fix the typo in my PR.

Reply:
#15939